core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -160,37 +163,40 @@ private void updateComputer(Node n, Map<String,Computer> byNameMap, Set<Computer
      * This method tries to reuse existing {@link Computer} objects
      * so that we won't upset {@link Executor}s running in it.
      */
-    protected void updateComputerList(boolean automaticSlaveLaunch) throws IOException {
```

Comment:
Can probably update `GetEnvironmentOutsideBuildTest` and `NodeProvisionerTest` and `QueueTest` accordingly.